packages/mui-joy/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -236,11 +236,10 @@ const ButtonGroup = React.forwardRef(function ButtonGroup(inProps, ref) {
           }
           const extraProps: Record<string, any> = {};
           if (isMuiElement(child, ['Divider'])) {
-            extraProps.inset = 'inset' in child.props ? child.props.inset : 'context';
+            extraProps.inset = (child.props as any).inset ?? 'context';
```

Comment:
I wonder if this would be better:

```suggestion
            const childProps = child.props as DividerProps;
            extraProps.inset = childProps.inset ?? 'context';
```

And reuse `childProps` for `orientation` below.

Reply:
Implemented 👌🏼